app/src/main/java/io/apicurio/registry/ibmcompat/api/impl/ApiServiceImpl.java

```diff
@@ -52,12 +77,147 @@
+            .collect(Collectors.toList());
+
+        schemaSummary.setId(schemaid);
+        schemaSummary.setName(schemaid); // TODO - add mechanism to store an artifact-level name
+
+        // The schema is disabled if all versions are disabled
+        boolean isSchemaDisabled = versionStates.stream().allMatch(versionState -> ArtifactState.DISABLED.equals(versionState));
+        schemaSummary.setEnabled(!isSchemaDisabled);
+
+        // The schema is deprecated if all versions are deprecated
+        boolean isSchemaDeprecated = versionStates.stream().allMatch(versionState -> ArtifactState.DEPRECATED.equals(versionState));
+        SchemaState schemaState = new SchemaState();
+        if(isSchemaDeprecated) {
+            schemaState.setState(SchemaState.StateEnum.DEPRECATED);
+        } else {
+            schemaState.setState(SchemaState.StateEnum.ACTIVE);
+        }
+        schemaSummary.setState(schemaState);
+    }
+
+    private void handleArtifactCreation(AsyncResponse response, String artifactId, String versionName, Throwable t) {
+        if (t != null) {
+            response.resume(t);
+            return;
+        }
+
+        if (!doesArtifactExist(artifactId)) {
+            // Some storage (such as asyncmem) needs a wait before the artifact is available for updating
+            try {
+                waitForArtifactCreation(artifactId);
+            } catch (ArtifactNotFoundException e) {
+                response.resume(e);
+                return;
+            }
+        }
+
+        try {
+            // Set the artifact name from the version name
+            EditableArtifactMetaDataDto dto = new EditableArtifactMetaDataDto();
+            dto.setName(versionName);
+            //
+            // TODO: On Kafka topic storage, this call to updateArtifactMetaData never returns.
+            // The consumer thread appears to be blocked. Help needed here.
+            //
+            // storage.updateArtifactMetaData(artifactId, dto);
+
+            // Prepare the response
+            SchemaInfo info = new SchemaInfo();
+            populateSchemaSummary(artifactId, info);
+            info.setVersions(getSchemaVersions(artifactId));
+
+            // updateArtifactMetaData call may be async, so also set the version name in the response
+            List<SchemaVersion> versions = info.getVersions();
+            versions.get(versions.size() - 1).setName(versionName);
+
+            response.resume(Response.status(Response.Status.CREATED).entity(info).build());
+        } catch (Throwable throwable) {
+            response.resume(throwable);
+        }
+    }
+
+    private void waitForArtifactCreation(String artifactId) throws ArtifactNotFoundException {
+        for (int i = 0; i < 5 && !doesArtifactExist(artifactId); i++) {
+            try {
+                Thread.sleep(200L);
+            } catch (InterruptedException e) {}
+        }
```

Comment:
Icky - at the very least needs to be configurable.  See my other comment on this.

Reply:
Uh ... only acceptable for tests, and even then try to avoid it as much as possible ...